crates/bevy_ecs/src/query/state.rs

```diff
@@ -220,6 +254,8 @@ where
         unsafe { self.iter_combinations_unchecked(world) }
     }
 
+    /// Iterates over all possible combinations of `K` query results for the given [`World`]
+    /// without repetition.
```

Comment:
this doc should be as explicit as `iter_combinations` (or link to it, but I would prefer to copy here for help in IDE) and mention why the `_mut`